### ISSUE TITLE
improve dht.put api

### ISSUE THREAD

**Issue by maxogden** [feross/bittorrent-dht#75]
im currently having to handle `dht.put` like this:

```js
    dht.put({v: new Buffer('hello world')}, function (errors, hash) {
      if (errors) {
        errors.forEach(function (e) {
          console.log(e)
        })
      }
      console.log(hash)
    })
```

however, I find the `errors` array not that useful for the following reasons:

- there will almost always be errors, since not all clients on the dht support bep44. for example:

```
{ [Error: 204 Unknown query type.] address: '95.211.211.152:6881' }
{ [Error: 202 Server Error] address: '121.40.163.12:6881' }
{ [Error: query timed out] address: '24.43.1.206:27935' }
{ [Error: query timed out] address: '105.107.66.81:8037' }
{ [Error: query timed out] address: '93.47.84.119:49670' }
{ [Error: query timed out] address: '145.167.133.38:32995' }
{ [Error: query timed out] address: '82.250.244.230:38754' }
{ [Error: query timed out] address: '5.79.212.119:62470' }
{ [Error: query timed out] address: '179.164.228.17:22828' }
{ [Error: query timed out] address: '27.208.33.164:9101' }
{ [Error: query timed out] address: '94.128.247.247:15718' }
{ [Error: query timed out] address: '81.246.190.143:3612' }
{ [Error: query timed out] address: '103.240.161.164:9097' }
{ [Error: query timed out] address: '172.101.33.37:18667' }
{ [Error: query timed out] address: '46.188.123.196:21985' }
```

- i have no way of knowing if any of the peers actually stored my key if all I get are the ones who didn't. IMO it should give you back all the successful ones too. there's no way to know how many peers it tried to store it on currently. 

other API suggestions welcome!

**Comment by mafintosh** [comment]
IMO think we should just have a single error if all the nodes failed and `null` otherwise. We could add an third argument with more info on how many nodes failed to support more advanced use cases

**Comment by maxogden** [comment]
@mafintosh +1

**Comment by feross** [comment]
Sounds fine to me. @substack?

**Comment by mafintosh** [comment]
fixed in 6.0.0. it'll only return a single error now if all puts failed. If a 300 error was returned (CAS mismiatch, bad seq) that one is picked - otherwise a generic "put failed" error is returned.
